services/QuillLMS/app/models/auth_credential.rb

```diff
@@ -24,4 +24,18 @@
 #
 class AuthCredential < ActiveRecord::Base
   belongs_to :user
+
+  def google_authorized?
+    provider == 'google' && refresh_token_valid?
```

Comment:
Let's make this `'google'` string a constant somewhere and start replacing occurrences of this string.